src/commands.rs

```diff
@@ -117,20 +115,23 @@ impl CommonOptions {
         };
         let config = self.read_config()?;
         get_signing_key(
-            &registry_url.map(|reg| reg.safe_label()),
-            config.keys.expect("Please set a default signing key by typing `warg key set <alg:base64>` or `warg key new"),
-            config.home_url,
+            registry_url.map(|reg| reg.safe_label()).as_deref(),
+            &config.keys.expect("Please set a default signing key by typing `warg key set <alg:base64>` or `warg key new"),
```

Comment:
Preexisting issue:

```suggestion
            &config.keys.context("please set a default signing key by typing `warg key set <alg:base64>` or `warg key new")?,
```

Ideally just `&config.keys` if we remove the wrapping `Option`.